app/javascript/hyacinth_ui_v1/components/projects/permissions/PermissionsEditor.js

```diff
@@ -0,0 +1,300 @@
+    },
+  );
+  const { loading: permissionsLoading, error: permissionsError } = useQuery(
+    getProjectPermissionsQuery, {
+      variables: { stringKey: projectStringKey },
+      onCompleted: (data) => {
+        setProjectPermissions(
+          collectionSortBy(data.projectPermissionsForProject, [perm => perm.user.sortName])
+        );
+      },
+    },
+  );
+
+  const [updateProjectPermissions, { error: updateProjectPermissionsError }] = useMutation(
+    updateProjectPermissionsMutation,
+  );
+
+  if (actionsLoading || permissionsLoading || usersLoading) return (<></>);
+  if (actionsError || permissionsError || usersError) {
+    return (
+      <GraphQLErrors errors={
+        actionsError || permissionsError || usersError || updateProjectPermissionsError
+      } />
+    );
+  }
+
+  const actionToDisplayLabel = (action) => {
+    return action.split('_').map(part => part.charAt(0).toUpperCase() + part.slice(1)).join(' ');
+  };
+
+  const updatePermissionsData = (userId, action, enabled) => {
+    setPermissionsChanged(true);
+
+    const { projectPermissionActions: {
+      actions: allActions, manageAction, actionsDisallowedForAggregatorProjects },
+    } = actionsData;
+
+    setProjectPermissions(
+      produce(projectPermissions, (draft) => {
+        const { actions: currentActionsForUser, project } = draft.find(
+          projectPermission => projectPermission.user.id === userId,
+        );
+        if (enabled) {
+          if (action === manageAction) {
+            // disable any current actions
+            currentActionsForUser.splice(0, currentActionsForUser.length);
+            // enable all allowed actions
+            if (project.isPrimary) {
+              currentActionsForUser.push(...allActions);
+            } else {
+              currentActionsForUser.push(...allActions.filter(
+                act => !actionsDisallowedForAggregatorProjects.includes(act)
+              ));
+            }
+          } else {
+            // enable single action
+            currentActionsForUser.push(action);
+          }
+        } else {
+          // disable single action
+          currentActionsForUser.splice(currentActionsForUser.indexOf(action), 1);
+        }
+      }),
+    );
+  };
+
+  const removeUser = (userId) => {
+    setPermissionsChanged(true);
+    setProjectPermissions(
+      produce(projectPermissions, (draft) => {
+        arrRemove(draft, (projectPermission) => {
+          return projectPermission.user.id === userId;
+        });
+      }),
+    );
+    setRemovedUserIds(
+      produce(removedUserIds, (draft) => {
+        draft.add(userId);
+      }),
+    );
+  };
+
+  const renderProjectPermission = (projectPermission) => {
+    const {
+      projectPermissionActions: {
+        actions, readObjectsAction, actionsDisallowedForAggregatorProjects,
+      },
+    } = actionsData;
+
+    return actions.map((action) => {
+      const disabledCheckbox = readOnly || action === readObjectsAction || (
```

Comment:
I would honestly be okay with not displaying the permissions that aren't available in aggregator projects.

Reply:
That would probably simplify things a bit.  I'll make that change, thanks!

Reply:
Rather than not displaying the permission boxes at all, I removed the checkboxes and I'm now displaying empty table cells.  I thought it might be a good reminder for users that these specific actions aren't available for aggregator projects, rather than hiding the columns altogether, and it'll also align more with what they eventually see in the user-view permissions editing context.  Hopefully that seems okay, but we can revisit this if you'd still prefer that the unavailable permission labels disappear.